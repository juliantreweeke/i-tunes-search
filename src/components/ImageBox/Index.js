import PropTypes from "prop-types";
import React from "react";
import LoadingSpinner from "../LoadingSpinner/Index";
import styles from "./imageBox.module.css";

const ImageBox = ({ image, alt, loading }) => (
  <>
    {loading ? (
      <LoadingSpinner />
    ) : (
      <div className={styles.imageBox}>
        <img className={styles.image} alt={alt} src={image} />
      </div>
    )}
  </>
);

ImageBox.propTypes = {
  data: PropTypes.object,
  loading: PropTypes.bool,
};

export default ImageBox;
