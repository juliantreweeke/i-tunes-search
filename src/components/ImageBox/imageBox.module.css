.spinnerContainer {
  display: flex;
  align-items: center;
  justify-content: center;
}

.imageBox {
  display: flex;
  height: 100%;
  flex-direction: column;
}

.imageBox .image {
  width: 100%;
  border-radius: var(--radius);
}
