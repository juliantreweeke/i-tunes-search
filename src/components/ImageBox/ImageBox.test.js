import React from "react";
import { render } from "@testing-library/react";
import ImageBox from "./Index";

describe("ImageBox", () => {
  const mockImageUrl = "test.com";
  const mockAlt = "altTag";

  it("should render", () => {
    const { container } = render(
      <ImageBox headingLevel="h1" image={mockImageUrl} alt={mockAlt} />
    );
    expect(container).toBeTruthy();
  });

  it("matches snapshot with prop data", () => {
    const { container } = render(
      <ImageBox headingLevel="h1" image={mockImageUrl} alt={mockAlt} />
    );
    expect(container).toMatchSnapshot();
  });

  it("matches snapshot with prop data when loading is true", () => {
    const { container } = render(
      <ImageBox
        headingLevel="h1"
        loading={true}
        image={mockImageUrl}
        alt={mockAlt}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
