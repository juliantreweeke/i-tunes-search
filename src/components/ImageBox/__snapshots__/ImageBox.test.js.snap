// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageBox matches snapshot with prop data 1`] = `
<div>
  <div
    class="imageBox"
  >
    <img
      alt="altTag"
      class="image"
      src="test.com"
    />
  </div>
</div>
`;

exports[`ImageBox matches snapshot with prop data when loading is true 1`] = `
<div>
  <div
    class="spinnerContainer"
  >
    <div
      class="spinner"
      id="loading-spinner"
    />
  </div>
</div>
`;
